src/quickpicks/schemas.ts

```diff
@@ -12,35 +12,11 @@ import { SchemaRegistry } from "../models/schemaRegistry";
  */
 export async function schemaSubjectQuickPick(
   schemaRegistry: SchemaRegistry,
-  onlyType: SchemaType | undefined = undefined,
   defaultSubject: string | undefined = undefined,
 ): Promise<string | undefined> {
   const loader = ResourceLoader.getInstance(schemaRegistry.connectionId);
-  const schemas = await loader.getSchemasForRegistry(schemaRegistry);
 
-  let schemaSubjects: string[] | undefined;
-
-  const latestVersionSchemaBySubject = new Map<string, Schema>();
-  if (schemas) {
-    // Crunch down to map of subject -> latest version'd Schema for said subject
-    for (const schema of schemas) {
-      // Skip if the caller asked for only a specific type of schema and this one doesn't match.
-      if (onlyType && schema.type !== onlyType) {
-        continue;
-      }
-      const latestVersionSchema = latestVersionSchemaBySubject.get(schema.subject);
-      if (!latestVersionSchema || schema.version > latestVersionSchema.version) {
-        latestVersionSchemaBySubject.set(schema.subject, schema);
-      }
-    }
-
-    schemaSubjects = Array.from(latestVersionSchemaBySubject.keys());
-    schemaSubjects.sort();
-  }
-
-  if (schemas === undefined) {
-    schemaSubjects = [];
-  }
+  const schemaSubjects = await loader.getSubjects(schemaRegistry);
```

Comment:
The core change right here. Basing `schemaSubjectQuickPick` on top of `loader.getSubjects(schemaRegistry)` returning the sorted subject names instead of `loader.getSchemasForRegistry(schemaRegistry)` returning all the schema metadatas.